pallets/moonbeam-orbiters/src/benchmarks.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+//! Benchmarking
+
+use crate::{BalanceOf, Call, Config, MinOrbiterDeposit, Pallet};
+use frame_benchmarking::{account, benchmarks, impl_benchmark_test_suite};
+use frame_support::traits::{Currency, Get, ReservableCurrency};
+use frame_system::RawOrigin;
+use sp_runtime::traits::StaticLookup;
+
+const MIN_ORBITER_DEPOSIT: u32 = 10_000;
+const USER_SEED: u32 = 999666;
+
+fn init<T: Config>() {
+	let min_orbiter_deposit: BalanceOf<T> = MIN_ORBITER_DEPOSIT.into();
+	MinOrbiterDeposit::<T>::put(min_orbiter_deposit);
+}
+
+/// Create a funded user.
+fn create_funded_user<T: Config>(string: &'static str, n: u32, balance: u32) -> T::AccountId {
+	const SEED: u32 = 0;
+	let user = account(string, n, SEED);
+	T::Currency::make_free_balance_be(&user, balance.into());
+	T::Currency::issue(balance.into());
+	user
+}
+
+/// Create a funded user and register it as a collator in orbiter program
+fn create_collator<T: Config>(string: &'static str, n: u32, balance: u32) -> T::AccountId {
+	let collator_account: T::AccountId = create_funded_user::<T>(string, n, balance);
+	let collator_lookup: <T::Lookup as StaticLookup>::Source =
+		T::Lookup::unlookup(collator_account.clone());
+	Pallet::<T>::add_collator(RawOrigin::Root.into(), collator_lookup)
+		.expect("fail to register collator");
+	collator_account
+}
+
+/// Create a funded user ard register it as an orbiter
+fn create_orbiter<T: Config>(string: &'static str, n: u32, balance: u32) -> T::AccountId {
+	let orbiter_account: T::AccountId = create_funded_user::<T>(string, n, balance);
+	Pallet::<T>::orbiter_register(RawOrigin::Signed(orbiter_account.clone()).into())
+		.expect("fail to register orbiter");
+	orbiter_account
+}
+
+benchmarks! {
+	collator_add_orbiter {
+		init::<T>();
+		let collator_account: T::AccountId = create_collator::<T>("COLLATOR", USER_SEED, 10_000);
+
+		// To test the worst case, we pre-fill the collator pool to the maximum size minus one
+		for i in 1..T::MaxPoolSize::get() {
```

Comment:
This will have to be re-run if MaxPoolSize ever increases.

If we intend to increase it often, a weight hint would be more appropriate. Adding a weight hint param also enables not charging worst case weight for every call.